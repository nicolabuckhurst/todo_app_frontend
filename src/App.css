
.myRow td,
.myRow td button {
    -webkit-transition: all 0.5s ease;  
    -moz-transition: all 0.5s ease;  
    -o-transition: all 0.5s ease;  
    -ms-transition: all 0.5s ease;  
    transition: all 0.5s ease;
}

.myRowAfterButtonClick,
.myRowAfterButtonClick td,
.myRowAfterButtonClick td button {
    font-size: 0px;
    margin: 0px;
    border: 0px;
    padding: 0px;
}