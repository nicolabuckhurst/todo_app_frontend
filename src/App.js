import React, { Component } from 'react';
import './App.css';
import Header from './components/header';
import Footer from './components/footer';
import Form from './components/form';
import Counter from './components/counter';
import Table from './components/table';
import {tasks, TasksContext} from './components/tasks-context'

class App extends Component {
  constructor(props){
    super(props);

    //create a counter for unique id's when new task is created..this is not a state as it doesn't need to be monitored for
    //change by react...we will not reuse id's when a task is deleted
    this.counter = 0;

    this.addTask = this.addTask.bind(this)
    this.toggleCompleteStatus = this.toggleCompleteStatus.bind(this)
    this.deleteTask = this.deleteTask.bind(this)

    //state initially contains an empty array for the tasks
    this.state = {tasks:[],
                  toggleCompleteStatus:this.toggleCompleteStatus,
                  deleteTask:this.deleteTask,
                  addTask:this.addTask,
                  }

  }


  //function for adding a new task and updating state
  addTask(taskText){
    console.log("called")
    //create a new uncompleted task based on taskText
    const task={
      id:this.counter,
      text:taskText,
      completed: false
    }
    //get current list of tasks stored in state and push new task onto end
    let taskList = this.state.tasks;
    taskList.push(task);
    //use setState to update the taskList stored in state
    this.setState({tasks:taskList})
    this.counter ++;
  }

  //function for changing task status to completed
  toggleCompleteStatus(id){
    //find task by id and toggle the completed property
    let taskList = this.state.tasks;
    for(let i=0; i<taskList.length; i++){
      if(taskList[i].id == id){
        taskList[i].completed = !taskList[i].completed //toggle the boolean value
      }
    }
    //use setState to update taskList stored in State
    this.setState({tasks:taskList})
  }

  //function to delete task
  deleteTask(id){
    //find task by id and delete from array
    let taskList = this.state.tasks;
    for(let i=0; i<taskList.length; i++){
      if(taskList[i].id == id){
        taskList.splice(i,1);
      }
    }
    //use setState to update taskList stored in State
    this.setState({tasks:taskList})
  }
  
  render() {
    return (
      <div style = {styles.transparentBox} className="d-flex flex-column">


        {/*wrap all the content except the footer in a div called "content"
        this makes the footer stick to the bottom
        of the viewport even when very little content on page
        there are no Bootstrap classes to do this automatically

        make the transparentBox that holds the content and the footer
        display flex (d-flex in bootstrap makes a container display flex and
        turns direct children into flex items)
        make the flex direction column so the "content" and "footer" are added one below the
        other just like block items so won't mess up any formatting

        make "content" flex-grow-1 so it grows in size to fill all the available vertical
        space in transparentBox (which is already set to height 100vh in css file)
        just leaving space for the footer*/}
            
        <div style={styles.content} className="pt-5 flex-grow-1"> {/*need padding as set navbar as fixed-top, makes it sit over the top of other content*/}
          
          <div className="container-fluid">
          <Header/>
            <TasksContext.Provider value={this.state}>
              <Form />
              <Counter />
              <Table toggleCompleteStatus={this.toggleCompleteStatus} deleteTask={this.deleteTask}/>
            </TasksContext.Provider>.
          </div> {/*end of container-fluid...bootstrap container for laying out components within main content of page*/}

        </div> {/*end of content div that resizes to whole of screen minus footer*/}

        <Footer/>
      </div> /*end of transparent box*/
    );
  }
}

const styles = {
  transparentBox:{
    height:"100vh"
  },
  content:{
    backgroundColor: "rgba(255,255,255,0.73)",
    overflowY: "auto"
  }
}

export default App;
