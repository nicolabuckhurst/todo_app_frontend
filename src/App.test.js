import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer'

//default test created by create-react-app 
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("the page renders correctly", ()=>{
  let component; //variable for holding testrenderer instance for App
  let tree; //variable to hold tree object created from testrenderer

  beforeAll(()=>{
     //create a test renderer instance with passed react element...doesn't use DOM but renders the component tree into memory
    component = renderer.create(<App />);
  })

  it('renders correctly onloading', ()=>{
    tree = component.toJSON();  //creates an object representing rendered tree..first time test is run this is used to create
                                //snapshot file
    expect(tree).toMatchSnapshot(); //subsequent time test is run will compare tree to snapshot..if you need to update snapshot
                                  //run jest -updatesnapshots MAKE SURE ANY FAILING SNAPSHOT TESTS DUE TO BUGS ARE FIXED FIRST
                                  //as ALL failing snapshots will be updated
  })

  it('renders correctly when a new task is added', ()=>{
    component.getInstance().addTask("feed dog"); //gets instance of root component (App.js) and manually trigger callback to add a new task
    tree = component.toJSON() //update tree object with newly rendered component and create a snapshot of 1st run
    expect(tree).toMatchSnapshot(); //compare tree with snapshot
  })  

  it('renders correctly when multiple tasks have been added', ()=>{
    component.getInstance().addTask("feed cat"); //gets instance of root component (App.js) and manually trigger callback to add a new task
    component.getInstance().addTask("feed bird");
    tree = component.toJSON() //update tree object with newly rendered component and create a snapshot of 1st run
    expect(tree).toMatchSnapshot(); //compare tree with snapshot
  })  
  
  it('renders correctly when task is completed', ()=>{
    component.getInstance().toggleCompleteStatus(0); //gets instance of root component (App.js) and manually trigger callback to add a new task
    tree = component.toJSON() //update tree object with newly rendered component and create a snapshot of 1st run
    expect(tree).toMatchSnapshot(); //compare tree with snapshot
  })  

  it('renders correctly when task is reinstated', ()=>{
    component.getInstance().toggleCompleteStatus(0); //gets instance of root component (App.js) and manually trigger callback to add a new task
    tree = component.toJSON() //update tree object with newly rendered component and create a snapshot of 1st run
    expect(tree).toMatchSnapshot(); //compare tree with snapshot
  })  

  it('renders correctly when task is deleted', ()=>{
    component.getInstance().deleteTask(0); //gets instance of root component (App.js) and manually trigger callback to add a new task
    tree = component.toJSON() //update tree object with newly rendered component and create a snapshot of 1st run
    expect(tree).toMatchSnapshot(); //compare tree with snapshot
  })  
  
})

