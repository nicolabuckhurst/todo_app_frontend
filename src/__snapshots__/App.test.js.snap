// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the page renders correctly renders correctly onloading 1`] = `
<div
  className="d-flex flex-column"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="pt-5 flex-grow-1"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.73)",
        "overflowY": "auto",
      }
    }
  >
     
    <div
      className="container-fluid"
    >
      <div
        className="row mb-3 mt-3"
      >
        <div
          className="col-12"
        >
          <p
            className="display-4 text-center"
          >
            To Do List
          </p>
        </div>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-row justify-content-center align-items-center mb-2 pb-4"
        >
          <div
            className="col-12 col-lg-8"
          >
            <div
              className="input-group mb-3"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="enter new task here"
                type="text"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-secondary"
                  type="submit"
                >
                  submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        className="row mb-4 justify-content-center"
      >
        <div
          className="col-10 text-center"
        >
          <button
            className="btn btn-lg"
            style={
              Object {
                "backgroundColor": "midnightblue",
                "color": "white",
              }
            }
            type="button"
          >
            Tasks Remaining: 
            <span
              className="badge badge-light ml-2"
            >
              0
            </span>
            <span
              className="sr-only"
            >
              tasks to do
            </span>
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <p
                className="h3"
              >
                Task List
              </p>
            </div>
          </div>
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <div
                className="table-responsive"
              >
                <table
                  className="table table-striped"
                >
                  <tbody />
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
      .
    </div>
     
  </div>
   
  <footer
    style={
      Object {
        "backgroundColor": "gray",
        "color": "white",
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 text-right mt-2 mb-2"
        >
          <a
            className="text-light"
            href="#"
          >
            Footer
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`the page renders correctly renders correctly when a new task is added 1`] = `
<div
  className="d-flex flex-column"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="pt-5 flex-grow-1"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.73)",
        "overflowY": "auto",
      }
    }
  >
     
    <div
      className="container-fluid"
    >
      <div
        className="row mb-3 mt-3"
      >
        <div
          className="col-12"
        >
          <p
            className="display-4 text-center"
          >
            To Do List
          </p>
        </div>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-row justify-content-center align-items-center mb-2 pb-4"
        >
          <div
            className="col-12 col-lg-8"
          >
            <div
              className="input-group mb-3"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="enter new task here"
                type="text"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-secondary"
                  type="submit"
                >
                  submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        className="row mb-4 justify-content-center"
      >
        <div
          className="col-10 text-center"
        >
          <button
            className="btn btn-lg"
            style={
              Object {
                "backgroundColor": "midnightblue",
                "color": "white",
              }
            }
            type="button"
          >
            Tasks Remaining: 
            <span
              className="badge badge-light ml-2"
            >
              1
            </span>
            <span
              className="sr-only"
            >
              tasks to do
            </span>
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <p
                className="h3"
              >
                Task List
              </p>
            </div>
          </div>
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <div
                className="table-responsive"
              >
                <table
                  className="table table-striped"
                >
                  <tbody>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed dog
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
      .
    </div>
     
  </div>
   
  <footer
    style={
      Object {
        "backgroundColor": "gray",
        "color": "white",
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 text-right mt-2 mb-2"
        >
          <a
            className="text-light"
            href="#"
          >
            Footer
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`the page renders correctly renders correctly when multiple tasks have been added 1`] = `
<div
  className="d-flex flex-column"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="pt-5 flex-grow-1"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.73)",
        "overflowY": "auto",
      }
    }
  >
     
    <div
      className="container-fluid"
    >
      <div
        className="row mb-3 mt-3"
      >
        <div
          className="col-12"
        >
          <p
            className="display-4 text-center"
          >
            To Do List
          </p>
        </div>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-row justify-content-center align-items-center mb-2 pb-4"
        >
          <div
            className="col-12 col-lg-8"
          >
            <div
              className="input-group mb-3"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="enter new task here"
                type="text"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-secondary"
                  type="submit"
                >
                  submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        className="row mb-4 justify-content-center"
      >
        <div
          className="col-10 text-center"
        >
          <button
            className="btn btn-lg"
            style={
              Object {
                "backgroundColor": "midnightblue",
                "color": "white",
              }
            }
            type="button"
          >
            Tasks Remaining: 
            <span
              className="badge badge-light ml-2"
            >
              3
            </span>
            <span
              className="sr-only"
            >
              tasks to do
            </span>
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <p
                className="h3"
              >
                Task List
              </p>
            </div>
          </div>
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <div
                className="table-responsive"
              >
                <table
                  className="table table-striped"
                >
                  <tbody>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed bird
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed cat
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed dog
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
      .
    </div>
     
  </div>
   
  <footer
    style={
      Object {
        "backgroundColor": "gray",
        "color": "white",
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 text-right mt-2 mb-2"
        >
          <a
            className="text-light"
            href="#"
          >
            Footer
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`the page renders correctly renders correctly when task is completed 1`] = `
<div
  className="d-flex flex-column"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="pt-5 flex-grow-1"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.73)",
        "overflowY": "auto",
      }
    }
  >
     
    <div
      className="container-fluid"
    >
      <div
        className="row mb-3 mt-3"
      >
        <div
          className="col-12"
        >
          <p
            className="display-4 text-center"
          >
            To Do List
          </p>
        </div>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-row justify-content-center align-items-center mb-2 pb-4"
        >
          <div
            className="col-12 col-lg-8"
          >
            <div
              className="input-group mb-3"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="enter new task here"
                type="text"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-secondary"
                  type="submit"
                >
                  submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        className="row mb-4 justify-content-center"
      >
        <div
          className="col-10 text-center"
        >
          <button
            className="btn btn-lg"
            style={
              Object {
                "backgroundColor": "midnightblue",
                "color": "white",
              }
            }
            type="button"
          >
            Tasks Remaining: 
            <span
              className="badge badge-light ml-2"
            >
              2
            </span>
            <span
              className="sr-only"
            >
              tasks to do
            </span>
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <p
                className="h3"
              >
                Task List
              </p>
            </div>
          </div>
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <div
                className="table-responsive"
              >
                <table
                  className="table table-striped"
                >
                  <tbody>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed bird
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed cat
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={
                          Object {
                            "color": "darkgray",
                            "textDecoration": "line-through",
                          }
                        }
                      >
                        feed dog
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Reinstate Task
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
      .
    </div>
     
  </div>
   
  <footer
    style={
      Object {
        "backgroundColor": "gray",
        "color": "white",
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 text-right mt-2 mb-2"
        >
          <a
            className="text-light"
            href="#"
          >
            Footer
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`the page renders correctly renders correctly when task is deleted 1`] = `
<div
  className="d-flex flex-column"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="pt-5 flex-grow-1"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.73)",
        "overflowY": "auto",
      }
    }
  >
     
    <div
      className="container-fluid"
    >
      <div
        className="row mb-3 mt-3"
      >
        <div
          className="col-12"
        >
          <p
            className="display-4 text-center"
          >
            To Do List
          </p>
        </div>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-row justify-content-center align-items-center mb-2 pb-4"
        >
          <div
            className="col-12 col-lg-8"
          >
            <div
              className="input-group mb-3"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="enter new task here"
                type="text"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-secondary"
                  type="submit"
                >
                  submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        className="row mb-4 justify-content-center"
      >
        <div
          className="col-10 text-center"
        >
          <button
            className="btn btn-lg"
            style={
              Object {
                "backgroundColor": "midnightblue",
                "color": "white",
              }
            }
            type="button"
          >
            Tasks Remaining: 
            <span
              className="badge badge-light ml-2"
            >
              2
            </span>
            <span
              className="sr-only"
            >
              tasks to do
            </span>
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <p
                className="h3"
              >
                Task List
              </p>
            </div>
          </div>
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <div
                className="table-responsive"
              >
                <table
                  className="table table-striped"
                >
                  <tbody>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed bird
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed cat
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
      .
    </div>
     
  </div>
   
  <footer
    style={
      Object {
        "backgroundColor": "gray",
        "color": "white",
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 text-right mt-2 mb-2"
        >
          <a
            className="text-light"
            href="#"
          >
            Footer
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`the page renders correctly renders correctly when task is reinstated 1`] = `
<div
  className="d-flex flex-column"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="pt-5 flex-grow-1"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.73)",
        "overflowY": "auto",
      }
    }
  >
     
    <div
      className="container-fluid"
    >
      <div
        className="row mb-3 mt-3"
      >
        <div
          className="col-12"
        >
          <p
            className="display-4 text-center"
          >
            To Do List
          </p>
        </div>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-row justify-content-center align-items-center mb-2 pb-4"
        >
          <div
            className="col-12 col-lg-8"
          >
            <div
              className="input-group mb-3"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="enter new task here"
                type="text"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-secondary"
                  type="submit"
                >
                  submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        className="row mb-4 justify-content-center"
      >
        <div
          className="col-10 text-center"
        >
          <button
            className="btn btn-lg"
            style={
              Object {
                "backgroundColor": "midnightblue",
                "color": "white",
              }
            }
            type="button"
          >
            Tasks Remaining: 
            <span
              className="badge badge-light ml-2"
            >
              3
            </span>
            <span
              className="sr-only"
            >
              tasks to do
            </span>
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <p
                className="h3"
              >
                Task List
              </p>
            </div>
          </div>
          <div
            className="row justify-content-center"
          >
            <div
              className="col-12 col-lg-8"
            >
              <div
                className="table-responsive"
              >
                <table
                  className="table table-striped"
                >
                  <tbody>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed bird
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed cat
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                    <tr
                      className="myRow"
                    >
                      <td
                        className="align-middle"
                        style={Object {}}
                      >
                        feed dog
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Done
                        </button>
                      </td>
                      <td
                        className="text-right"
                        style={
                          Object {
                            "width": "8%",
                          }
                        }
                      >
                        <button
                          className="btn btn-secondary btn-sm"
                          onClick={[Function]}
                          type="button"
                        >
                          Delete
                        </button>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
      .
    </div>
     
  </div>
   
  <footer
    style={
      Object {
        "backgroundColor": "gray",
        "color": "white",
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 text-right mt-2 mb-2"
        >
          <a
            className="text-light"
            href="#"
          >
            Footer
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
